### PR TITLE
SIMSBIOHUB-13: Lock Submitted Data and Files

### PR DESCRIPTION
**Pull request by KjartanE**
# Overview

## Links to Jira tickets

- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-13

## Description of relevant changes

- {List all relevant changes, in particular anything that will help the reviewers test/verify this PR}

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

**Comment by NickPhura** [comment]
@KjartanE handful of bugs/code smells to address

File: api/src/services/project-service.ts [hunk 1]
```diff
@@ -688,36 +689,55 @@ export class ProjectService extends DBService {
    * @param {number} projectId
    * @returns {*} {Promise<boolean>}
```
Comment (NickPhura) [inline api/src/services/project-service.ts]: nit: JSDoc is out of date

File: api/src/services/project-service.ts [hunk 2]
```diff
@@ -688,36 +689,55 @@ export class ProjectService extends DBService {
...
+
+    return PublishStatus.SUBMITTED;
   }
 
   /**
-   * Returns true if any surveys for a given project have unpublished data or false if all survey under a project are published
+   * Returns publish status of a given project
    *
    * @param projectId
    * @returns {*} {Promise<boolean>}
```
Comment (NickPhura) [inline api/src/services/project-service.ts]: nit: JSDoc

**Comment by NickPhura** [comment]
@KjartanE One issue I found: 

Setup:
- I have a project with 1 survey (no other attachments)
- The survey has an observation uploaded (no other attachments).
- I submit the observation to biohub, successfully.

Result:
- The survey page banner/chip shows it is submitted (green) ✔️ 
- The project page banner/chip still shows it is unsubmitted, including the survey list chip (grey) ❌ 
- The project list page shows it is unsubmitted (grey) ❌

File: api/src/repositories/summary-repository.ts [hunk 3]
```diff
@@ -299,11 +301,11 @@ export class SummaryRepository extends BaseRepository {
 
     const response = await this.connection.query<{ delete_timestamp: string }>(sqlStatement.text, sqlStatement.values);
 
-    if (!response || !response?.rows[0]?.delete_timestamp) {
+    if (!response?.rows[0]?.delete_timestamp) {
       throw new HTTP400('Failed to soft delete survey summary submission');
     }
 
-    return (response && response.rowCount) || null;
+    return response?.rowCount || null;
```
Comment (NickPhura) [inline api/src/repositories/summary-repository.ts:308]: super nit: technically we don't need to `?.` the rows or rowCount variable, as they will always be present in the response object.

Comment (NickPhura) [reply api/src/repositories/summary-repository.ts:308]: (The sonarcloud thing isnt smart enough to know that, it just sees the `||` check and raises the warning that it prefers `?.`)

**Review by NickPhura — APPROVED** [review]
Tested locally, including submitting observations to biohub.

**Review by AlfredRosenthal — APPROVED** [review]
🐶

**Pull request merged**